prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1066 @@
+                        ReportLineage(
+                            taskIndex,
+                            actionId,
+                            report.id,
+                            childReport.id,
+                            OffsetDateTime.now()
+                        ),
+                        txn
+                    )
+            }
+
+            ReportStreamTestDatabaseContainer.testDatabaseAccess.insertTask(
+                report,
+                fileFormat.toString().lowercase(),
+                report.bodyURL,
+                nextAction = ProcessEvent(
+                    nextEventAction,
+                    report.id,
+                    Options.None,
+                    emptyMap(),
+                    emptyList()
+                ),
+                txn
+            )
+        }
+
+        return report
+    }
+
+    private fun createReportsWithLineage(
+        reportContents: String,
+        topic: Topic = Topic.FULL_ELR,
+    ): Pair<Report, Report> {
+        val receivedBlobUrl = BlobAccess.uploadBlob(
+            "receive/mr_fhir_face.fhir",
+            reportContents.toByteArray(),
+            getBlobContainerMetadata()
+        )
+
+        val convertedBlobUrl = BlobAccess.uploadBlob(
+            "convert/mr_fhir_face.fhir",
+            reportContents.toByteArray(),
+            getBlobContainerMetadata()
+        )
+
+        val convertReport = createReport(
+            Report.Format.FHIR,
+            TaskAction.convert,
+            TaskAction.route,
+            Event.EventAction.ROUTE,
+            topic,
+            0,
+            null,
+            convertedBlobUrl
+        )
+
+        val receiveReport = createReport(
+            Report.Format.FHIR,
+            TaskAction.receive,
+            TaskAction.convert,
+            Event.EventAction.CONVERT,
+            topic,
+            0,
+            convertReport,
+            receivedBlobUrl
+        )
+
+        return Pair(receiveReport, convertReport)
+    }
+
+    private fun checkActionTable(expectedTaskActions: List<TaskAction>) {
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val actionRecords = DSL.using(txn)
+                                   .select(Tables.ACTION.asterisk())
+                                   .from(Tables.ACTION)
+                                   .fetchInto(
+                                       Action::class.java
+                                   )
+
+            for (i in 0 until actionRecords.size) {
+                assertEquals(expectedTaskActions.get(i), actionRecords.get(i).actionName)
+            }
+        }
+    }
+
+    private data class ActionLogRecordContent (
+        val index: Int,
+        val reportId: UUID?,
+        val type: ActionLogLevel?,
+        val scope: ActionLogScope?
+    )
+
+    private val expectedButStrangeObservationFilterEntry = listOf(
+        ActionLogRecordContent(
+            0,
```

Comment:
Nit: could this get tweaked a little to reflect the fact that the index is not zero based.  It's already confusing to begin and I think this adds to the confusion some.